### PR TITLE
chore(bot-v3): nodejs bot plugin v3

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

**Comment by jayzhang** [comment]
1. create Nodejs bot plugin for v3
2. refactor: merge local provision and remote provision into one API

File: packages/fx-core/src/plugins/resource/bot/v3/index.ts [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+  UnzipError,
+} from "./error";
+import { ensureSolutionSettings } from "../../../solution/fx-solution/utils/solutionSettingsHelper";
+
+@Service(BuiltInFeaturePluginNames.bot)
+export class NodeJSBotPluginV3 implements v3.FeaturePlugin {
+  name = BuiltInFeaturePluginNames.bot;
+  displayName = "NodeJS Bot";
+  async pluginDependencies(ctx: v2.Context, inputs: Inputs): Promise<Result<string[], FxError>> {
+    return ok([BuiltInFeaturePluginNames.identity]);
```
Comment (IvanJobs) [inline packages/fx-core/src/plugins/resource/bot/v3/index.ts:84]: Mark: any background about this method?

Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/bot/v3/index.ts:84]: Bot plugin depends on identity, if identity is missing, the arm template will have errors.
Declaring the dependency here means: if bot is added, identity plugin should also be added together.

**Review by IvanJobs — APPROVED** [review]
lgtm for the bot part.

**Pull request merged**